python/dispatcher/dispatcher.py

```diff
@@ -0,0 +1,119 @@
+# -------------------------------------------------------------------------
+#
+# Copyright (c) 2024 General Motors GTO LLC
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# SPDX-FileType: SOURCE
+# SPDX-FileCopyrightText: 2024 General Motors GTO LLC
+# SPDX-License-Identifier: Apache-2.0
+#
+# -------------------------------------------------------------------------
+
+
+import socket
+from typing import Dict
+from threading import Thread
+from threading import Lock
+import uuid
+import logging 
+import sys
+
+
+PORT = 44444
+IP = "127.0.0.1" 
+ADDR = (IP, PORT)
+
+logging.basicConfig(format='%(asctime)s %(message)s')
+# create logger
+logger = logging.getLogger('simple_example')
+logger.setLevel(logging.INFO)
+
+
+# Holds all client connections
+clients: Dict[str, socket.socket] = {}
+# Create the shared lock
+lock = Lock()
+
+def close_socket(client_id: str, locket: Lock):
```

Comment:
Here is an optimized version. 

```python
def close_socket(client_id: str):
    with client_lock:
        client_socket = clients.pop(client_id, None)
        if client_socket:
            client_socket.close()
```


 It is more robust when dealing with potential edge cases where the client ID might not exist in the dictionary.